### ISSUE TITLE
vanX: `Cannot read properties of null (reading 'Symbol()')` issue when `replace`ing an array of non-null objs with partially-null fields

### ISSUE THREAD

**Issue by metaleap** [vanjs-org/van#150]
Bear with me if I paste some slightly verbose (tho still trim & banal) code... the array's item type being `reactive([])`d here is this:

```typescript
export type Post = {
	Id: number
	DtMade?: string
	DtMod?: string
	By: number // Id of User
	Files: string[]
	Md: string // text content of post
	Repl: number // Id of another Post, or 0
	To: number[] // Id of User(s)
}
```

(This is in a codegenerated-from-my-API `.ts`, no `van` let alone state fields here as you can see. Simple as it gets objs.)

This simplistic / prototypal component fails during `replace` as will be stacktrace-detailed right afterwards:

```typescript
import van from '../../__yostatic/vanjs/van-1.2.3.debug.js'
import * as vanx from '../../__yostatic/vanjs/van-x.js'
import * as yo from '../yo-sdk.js'

const htm = van.tags

export type UiCtlPosts = {
    DOM: HTMLElement
    posts: vanx.Reactive<yo.Post[]>
    update: (_: yo.Post[]) => void
}

export function create(): UiCtlPosts {
    const me: UiCtlPosts = {
        DOM: htm.div({ 'class': 'haxsh-posts' }),
        posts: vanx.reactive([] as yo.Post[]),
        update: (posts) => update(me, posts),
    }
    van.add(me.DOM, vanx.list(htm.div, me.posts, (it) => {
        return htm.div({}, it.val.Md)
    }))
    return me
}

function update(me: UiCtlPosts, posts: yo.Post[]) {
    vanx.replace(me.posts, (_: yo.Post[]) => posts)
}
```

Now here's what happens on every `update` call — I have console-verified that `posts` is always filled with 123 non-`null` `Post` objects. No fields are ever `null` except occasionally `To`. And what is `catch`ed is this:

```json
{
  "message":"Cannot read properties of null (reading 'Symbol()')",
  "stack"  :"TypeError: Cannot read properties of null (reading 'Symbol()')
    at toState (http://localhost:5252/__yostatic/vanjs/van-x.js:11:21)
    at http://localhost:5252/__yostatic/vanjs/van-x.js:16:74
    at Array.map (<anonymous>)
    at reactive (http://localhost:5252/__yostatic/vanjs/van-x.js:16:54)
    at toState (http://localhost:5252/__yostatic/vanjs/van-x.js:11:72)
    at http://localhost:5252/__yostatic/vanjs/van-x.js:96:25
    at Array.map (<anonymous>)
    at Module.replace (http://localhost:5252/__yostatic/vanjs/van-x.js:94:38)
    at update (http://localhost:5252/__static/ui/posts.js:26:8)
    at Object.update (http://localhost:5252/__static/ui/posts.js:12:24)"
}
```

To circumvent this, I have found that this alternative `update` impl would be necessary instead:

```typescript
function update(me: UiCtlPosts, posts: yo.Post[]) {
    vanx.replace(me.posts, (_: yo.Post[]) => posts.map(it => {
        if (!it.To)
            it.To = []
        return it
    }))
}
```

Imho shouldn't be necessary, right? Might have minified away or forgotten a `?` in the still-new, otherwise funky and already-greatly-appreciated vanX perhaps  =)

**Comment by Tao-VanJS** [comment]
Thanks for reporting the bug! I think it should be fixed in VanX `0.1.2` release: https://github.com/vanjs-org/van/commit/0bb8665237e28bab6f369393abb40d8602a2976e#diff-013e51e8f53fd21cce723bb5e249a9676facefb8f2b6f2a9eac2b2b3b9ccba30.

Could you check if things are working on your side with VanX `0.1.2`?

**Comment by metaleap** [comment]
Good going! 0.1.2 did fix this afaict. Very nice, muchos gracias!
